pkg/scheduler/internal/cache/cache.go

```diff
@@ -238,6 +238,14 @@ func (cache *schedulerCache) UpdateNodeInfoSnapshot(nodeSnapshot *schedulernodei
 			}
 		}
 	}
+
+	// Take a snapshot of the nodes order in the tree
+	nodeSnapshot.NodeInfoList = make([]*schedulernodeinfo.NodeInfo, 0, cache.nodeTree.numNodes)
+	for i := 0; i < cache.nodeTree.numNodes; i++ {
+		if n := nodeSnapshot.NodeInfoMap[cache.nodeTree.next()]; n != nil {
```

Comment:
Do we know under what circumstances `n` can be nil? Not sure if we should log something if it happens.

Reply:
inconsistency between nodeTree and nodeInfoMap, I will log an error.

Reply:
done.